android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java

```diff
@@ -247,6 +300,56 @@ public void onClick(View v) {
                 }
             });
         }
+
+        btnBuySendSwap.setOnClickListener(v -> {
+            if (mActivityType == ActivityType.SEND) {
+                // TODO(sergz): token transfers via a token contract
+                String to = toValueText.getText().toString();
+                if (to.isEmpty()) {
+                    // TODO(sergz): some address validation
+                    return;
+                }
+                TextView fromValueText = findViewById(R.id.from_value_text);
+                String value = fromValueText.getText().toString();
+                // TODO(sergz): Some kind of validation that we have enough balance
+                // TODO(sergz): Get gas price
+                TxData data = Utils.getTxData("0x1", "0x2FBF9BEDA5F0", "0x2FBF9BEDA5F1", to,
+                        Utils.toHexWei(value), new byte[0]);
+                Spinner accountSpinner = findViewById(R.id.accounts_spinner);
+                String from = mCustomAccountAdapter.getTitleAtPosition(
+                        accountSpinner.getSelectedItemPosition());
+                if (mEthTxController != null) {
+                    mEthTxController.addUnapprovedTransaction(data, from,
+                            (success, tx_meta_id)
+                                    -> {
+                                            // TODO(sergz): Add an observer and approve transaction
```

Comment:
https://github.com/brave/brave-browser/issues/18076

Reply:
btw here I don't think you'll do anything. I think you want the observer to be independent of this and respond by opening the UI to approve. When the UI action for approval happens you can call to approve the transaction.

Reply:
yep, just posted it somewhere to not forget